zuul-core/src/main/java/com/netflix/netty/common/proxyprotocol/StripUntrustedProxyHeadersHandler.java

```diff
@@ -96,11 +106,20 @@ private boolean connectionIsUsingMutualSSLWithAuthEnforced(Channel ch)
         return is;
     }
 
-    private void stripXFFHeaders(HttpRequest req)
+    @VisibleForTesting
+    void stripXFFHeaders(HttpRequest req)
     {
         HttpHeaders headers = req.headers();
         for (AsciiString headerName : HEADERS_TO_STRIP) {
             headers.remove(headerName);
         }
     }
+
+    @VisibleForTesting
+    void checkBlacklist(HttpRequest req, List<String> blacklist) {
+        // blacklist headers from
+        if (blacklist.contains(req.headers().get(HttpHeaderNames.HOST).toLowerCase())) {
```

Comment:
There should be a `normalized()` method you can call that avoids the lowercase string allocation.

Reply:
Unfortunately it's a Netty header object and not ours. Do you know if they handle casing on gets? If they do, we can just remove the line.

Reply:
Hmm, I guess not.  consider using equalsIgnoreCase to avoid the alloc.